wal/wal.go

```diff
@@ -332,6 +332,7 @@ func (w *WAL) Repair(origErr error) error {
 		return err
 	}
 	w.segment = s
+	w.donePages = 0
```

Comment:
how about instead of this add an internal func like w.setSegment to avoid this bug again.